From 721fbac99c49ebe23a564726e753d8b581441a09 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Wed, 21 May 2025 06:50:00 +0000
Subject: [PATCH] Fix the fowwlowing build error :

include/linux/types.h:35:33: error: 'bool' cannot be defined via 'typedef'
   35 | typedef _Bool                   bool;

Upstream-Status: Pending
Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 include/linux/stdarg.h | 10 ++++++++++
 include/linux/stddef.h |  2 ++
 include/linux/types.h  |  2 ++
 3 files changed, 14 insertions(+)

diff --git a/include/linux/stdarg.h b/include/linux/stdarg.h
index c8dc7f4f390c..af7eb0ed138e 100644
--- a/include/linux/stdarg.h
+++ b/include/linux/stdarg.h
@@ -3,7 +3,17 @@
 #define _LINUX_STDARG_H
 
 typedef __builtin_va_list va_list;
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ > 201710L
+#define va_start(v, ...) __builtin_va_start(v, 0)
+#ifdef __has_builtin
+#if __has_builtin(__builtin_c23_va_start)
+#undef va_start
+#define va_start(...)	__builtin_c23_va_start(__VA_ARGS__)
+#endif
+#endif
+#else
 #define va_start(v, l)	__builtin_va_start(v, l)
+#endif
 #define va_end(v)	__builtin_va_end(v)
 #define va_arg(v, T)	__builtin_va_arg(v, T)
 #define va_copy(d, s)	__builtin_va_copy(d, s)
diff --git a/include/linux/stddef.h b/include/linux/stddef.h
index 929d67710cc5..1b796b519519 100644
--- a/include/linux/stddef.h
+++ b/include/linux/stddef.h
@@ -7,10 +7,12 @@
 #undef NULL
 #define NULL ((void *)0)
 
+#if !defined(__STDC_VERSION__) || __STDC_VERSION__ < 202311
 enum {
 	false	= 0,
 	true	= 1
 };
+#endif
 
 #undef offsetof
 #define offsetof(TYPE, MEMBER)	__builtin_offsetof(TYPE, MEMBER)
diff --git a/include/linux/types.h b/include/linux/types.h
index 78d87c751ff5..9206f802e6dd 100644
--- a/include/linux/types.h
+++ b/include/linux/types.h
@@ -32,7 +32,9 @@ typedef __kernel_timer_t	timer_t;
 typedef __kernel_clockid_t	clockid_t;
 typedef __kernel_mqd_t		mqd_t;
 
+#if !defined(__STDC_VERSION__) || __STDC_VERSION__ < 202311L
 typedef _Bool			bool;
+#endif
 
 typedef __kernel_uid32_t	uid_t;
 typedef __kernel_gid32_t	gid_t;
-- 
2.43.0

